config/src/keys.rs

```diff
@@ -148,15 +163,28 @@ impl NetworkKeyPairs {
 // ConsensusKeyPair is used to store a validator's consensus keypair.
 // It is filled via a config file at the moment.
 #[derive(Debug, Deserialize, PartialEq, Serialize)]
-#[cfg_attr(any(test, feature = "fuzzing"), derive(Clone))]
 pub struct ConsensusKeyPair {
     consensus_private_key: PrivateKeyContainer<Ed25519PrivateKey>,
     #[serde(serialize_with = "serialize_opt_key")]
     #[serde(deserialize_with = "deserialize_opt_key")]
     consensus_public_key: Option<Ed25519PublicKey>,
 }
 
-// required for serialization
+impl Clone for ConsensusKeyPair {
+    #[cfg(any(test, feature = "fuzzing"))]
+    fn clone(&self) -> Self {
+        Self {
+            consensus_private_key: self.consensus_private_key.clone(),
+            consensus_public_key: self.consensus_public_key.clone(),
+        }
+    }
+
+    #[cfg(not(any(test, feature = "fuzzing")))]
+    fn clone(&self) -> Self {
+        Self::default()
+    }
+}
+
```

Comment:
Same for this. I really worry that we'll just end up operating with invalid keys when we least expect it.